adam/ontology/ontology.py

```diff
@@ -152,8 +158,36 @@ def nodes_with_properties(
             if self.has_all_properties(
                 node, required_properties, banned_properties=banned_properties
             )
+            and node not in banned_ontology_types
+            and not self.descends_from(node, banned_ontology_types)
         )
 
+    def descends_from(
```

Comment:
@spigo900 : Please add a unit test for this.

Reply:
@gabbard Done, [commit here](https://github.com/isi-vista/adam/pull/773/commits/91d5bb7bb9c2e0a0e8679e52618f2468be76d391). Does it make sense to use the phase 1 ontology in these tests, or should I rewrite them to use an artificial ontology?

Reply:
You can use Phase 1